mod-source-record-manager-server/src/main/java/org/folio/verticle/AbstractConsumersVerticle.java

```diff
@@ -30,41 +31,45 @@ public abstract class AbstractConsumersVerticle extends AbstractVerticle {
   @Value("${srm.kafka.DataImportConsumer.loadLimit:5}")
   private int loadLimit;
 
-  private List<KafkaConsumerWrapper<String, String>> consumerWrappersList = new ArrayList<>();
+  @Autowired
+  private KafkaConsumersStorage kafkaConsumersStorage;
 
   @Override
   public void start(Promise<Void> startPromise) {
     context.put("springContext", springGlobalContext);
 
     SpringContextUtil.autowireDependencies(this, context);
 
+    List<Future<Void>> futures = new ArrayList<>();
+
     getEvents().forEach(event -> {
       SubscriptionDefinition subscriptionDefinition = KafkaTopicNameHelper
         .createSubscriptionDefinition(kafkaConfig.getEnvId(),
           KafkaTopicNameHelper.getDefaultNameSpace(),
           event);
-      consumerWrappersList.add(KafkaConsumerWrapper.<String, String>builder()
+      KafkaConsumerWrapper<String, String> consumerWrapper = KafkaConsumerWrapper.<String, String>builder()
         .context(context)
         .vertx(vertx)
         .kafkaConfig(kafkaConfig)
         .loadLimit(loadLimit)
         .globalLoadSensor(globalLoadSensor)
         .subscriptionDefinition(subscriptionDefinition)
         .processRecordErrorHandler(getErrorHandler())
-        .build());
-    });
-    List<Future<Void>> futures = new ArrayList<>();
-    consumerWrappersList.forEach(consumerWrapper ->
+        .backPressureGauge(getBackPressureGauge())
+        .build();
+      kafkaConsumersStorage.addConsumer(event, consumerWrapper);
```

Comment:
Curious how consumers in other instances of SRM play with each other. Thinking maybe there might be some rogue instance of SRM still feeding large data import events instead of letting OCLC imports through.

Reply:
For now flow control intended for OCLC imports, but design should be extendable for future. For example, Spitfire team going to use our DI flow for importing quick marcs. This quick marcs imports should also be processed in between of large data imports. To use the same mechanism quick marc app should put messages in subsequent topic in flow after DI_RAW_RECORDS_CHUNK_READ. This tactics will allow to import quick marcs quickly without any modifications from our side.

Reply:
let me ask a different way.

Two instances of SRM are running. Both have a consumer defined on a topic to read from it. Would there be situations where both instances of SRM will try to pause the topic at the same time?

Reply:
For example, we have 2 instances of SRM. Each instance configured to have 3 consumers in consumer group to read from DI_RAW_RECORDS_CHUNK_READ topic by default.
Each instance would have separate counter to track internal state of flow control. If for example max.simultaneous.records met for instance №1 - it will be paused, but 3 consumers from instance №2 would continue reading.
We pausing only consumers, producers always working.

Reply:
ok so the consumers are being paused, not the topic. I knew this but i was not using it in my thought process. Thanks for the clarification!